src/chat/chat.go

```diff
@@ -47,9 +63,27 @@ func sendMessage(message string) {
 	}
 }
 
-// Helper function to retreive text message from console
-func getChatMessage() string {
-	reader := bufio.NewReader(os.Stdin)
-	text, _ := reader.ReadString('\n')
-	return strings.TrimSpace(text)
+// Web-based chat message sending
+func SetupWebService() {
+
+	if _, err := strconv.Atoi(*chatPort); err != nil {
+		panic("Invalid chat port specified (must be 32-bit integer). You put: " + *chatPort)
+	}
+
+	http.HandleFunc(chatResource, chatHandler)
+	http.ListenAndServe(":"+*chatPort, nil)
```

Comment:
Move the ListenAndServe call to main(), invoking it with go http.ListenAndServe(...) after chat.SetupWebService has run. Since SetupWebService no longer has a blocking call at the end of it, there's no need to launch it on a separate thread (and in fact, it is preferable not to do so to ensure http.HandleFunc has been run prior to ListenAndServer.)